src/lib/loader.c

```diff
@@ -262,6 +265,13 @@ static void __attribute__((constructor)) loader() {
   ensure_libpthread_is_loaded();
   ensure_librt_is_loaded();
 
+  // now that we have loaded libdl, we can ensure that we use the real
+  // dlopen function (instead of internal libc function)
+  if (s_libdl_handle) {
```

Comment:
Code could be moved to `ensure_libdl_is_loaded`, `s_dlerror` should be updated only if null